### DISCUSSION TITLE
RKE2 support for KUBE OVN CNI

### DISCUSSION THREAD

**Discussion by sureshgoli25** [rancher/rke2#4649] [Ideas]
Hi, We are looking forward to see the support of Kube OVN CNI by RKE2.

Kube OVN got many features like dedicated subnets for namespaces, isolated subnets, Multi-Cluster Networking, egress gateway which supports many use cases are looking forward.

Since RKE2 is mainly focused on security, it might be good idea to include this CNI as well.

**Comment by manuelbuil** [comment]
Yeah, OVN CNI is a great CNI plugin, however, we are quite overloaded with tasks right now. Would you be willing to contribute it?

**Comment by uliah-h** [comment]
There is a way to deploy.

-  install the rke2-server  with `--cni none`

```
tee -a /etc/rancher/rke2/config.yaml <<EOF
cni:
  - none
EOF 

systemctl start rke2-server
```

- install KUBE-OVN

> [KUBE-OVN](https://kubeovn.github.io/docs/v1.12.x/start/one-step-install/#helm-chart)  Helm Chart installation is supported since Kube-OVN v1.12.0, and the default deployment is Overlay networking. 
```
kubectl label node -lnode-role.kubernetes.io/control-plane kube-ovn/role=master --overwrite

tee values.yaml <<EOF
networking:
  NET_STACK: ipv4
ipv4:
  POD_CIDR: "10.42.0.0/16"
  POD_GATEWAY: "10.42.0.1"
  SVC_CIDR: "10.43.0.0/16"
  JOIN_CIDR: "100.64.0.0/16"
EOF

helm -n kube-system install kube-ovn kubeovn/kube-ovn --set MASTER_NODES=${Node0IP} -f values.yaml
```
I run a sample app on Kubernetes ,it's ok

![image](https://github.com/rancher/rke2/assets/37589037/4b2a01f6-73f0-48dd-8c5c-60a397e850d9)

**Comment by pratik705** [comment]
in my lab if i set cni to none, nodes doesnt move to ready state. I can see following error:
`container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized`

@FruitH Am I missing anything here?

**Reply by brandond** [reply]
You're missing the step where you deploy the CNI?

**Reply by pratik705** [reply]
@brandond I am deploying rke2 with "cni: none"[1] as I want to use kube-ovn CNI. I was going to install kube-ovn once the kubernetes cluster is up. But, looks like as the CNI is not present during the installation, the nodes will not be in ready state[2]. How I can make kube-ovn available during the cluster installation? 


[1]
```
# cat /etc/rancher/rke2/config.yaml

server: https://xx.xx.xx.xx:9345
token: defaultSecret12345
data-dir: /var/lib/rancher/rke2
cni: none
```


[2]
```
NAME       STATUS
node01   NotReady
node02   NotReady
node03   NotReady
```

**Reply by pratik705** [reply]
I managed to deploy kube-ovn. Thanks for the response :-)
